logos/logos-agent/app/triggers.py

```
@@ -774,6 +899,22 @@ async def _may_direct_changes(self, comments: list[dict[str, Any]]) -> bool:
 
+    async def _pull_request(self, number: int) -> dict[str, Any] | None:
+        """That number as a pull request, or None when it is an issue.
+
+        Cached for the pass: one thread asks about one number, and a poll
+        that answered three questions about the same pull request would
+        otherwise ask GitHub three times.
+        """
+        if number not in self._pulls:
+            try:
+                self._pulls[number] = await github.pull_request(number) or None
+            except Exception:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)